src/Altinn.App.Core/Internal/Process/ProcessEngine.cs

```diff
@@ -1,138 +1,289 @@
 {
+    private readonly IProcessReader _processReader;
+    private readonly IProfile _profileService;
+    private readonly IProcessNavigator _processNavigator;
+    private readonly IProcessEventDispatcher _processEventDispatcher;
+
     /// <summary>
-    /// The process engine is responsible for all BMPN related functionality
-    ///
-    /// It will call processChange handler that is responsible
-    /// for the business logic happening for any process change.
+    /// Initializes a new instance of the <see cref="ProcessEngine"/> class
     /// </summary>
-    public class ProcessEngine : IProcessEngine
+    /// <param name="processReader"></param>
+    /// <param name="profileService"></param>
+    /// <param name="processNavigator"></param>
+    /// <param name="processEventDispatcher"></param>
+    public ProcessEngine(
+        IProcessReader processReader,
+        IProfile profileService,
+        IProcessNavigator processNavigator,
+        IProcessEventDispatcher processEventDispatcher)
     {
-        private readonly IProcessChangeHandler _processChangeHandler;
-
-        private readonly IProcessReader _processReader;
-        private readonly IFlowHydration _flowHydration;
-
-        /// <summary>
-        /// Initializes a new instance of the <see cref="ProcessEngine"/> class.
-        /// </summary>
-        public ProcessEngine(
-                IProcessChangeHandler processChangeHandler,
-                IProcessReader processReader,
-                IFlowHydration flowHydration)
-        {
-            _processChangeHandler = processChangeHandler;
-            _processReader = processReader;
-            _flowHydration = flowHydration;
-        }
+        _processReader = processReader;
+        _profileService = profileService;
+        _processNavigator = processNavigator;
+        _processEventDispatcher = processEventDispatcher;
+    }
 
-        /// <summary>
-        /// Move process to next element in process
-        /// </summary>
-        public async Task<ProcessChangeContext> Next(ProcessChangeContext processChange)
+    /// <inheritdoc/>
+    public async Task<ProcessChangeResult> StartProcess(ProcessStartRequest processStartRequest)
+    {
+        if (processStartRequest.Instance.Process != null)
         {
-            string? currentElementId = processChange.Instance.Process.CurrentTask?.ElementId;
-
-            if (currentElementId == null)
+            return new ProcessChangeResult()
             {
-                processChange.ProcessMessages = new List<ProcessChangeInfo>();
-                processChange.ProcessMessages.Add(new ProcessChangeInfo() { Message = $"Instance does not have current task information!", Type = "Conflict" });
-                processChange.FailedProcessChange = true;
-                return processChange;
-            }
+                Success = false,
+                ErrorMessage = "Process is already started. Use next.",
+                ErrorType = "Conflict"
```

Comment:
Consider creating a ProcessErrorType class holding string constants to avoid magic strings as the same type is repeated.